test/test_type_promotion.py

```diff
@@ -593,6 +593,38 @@ def test_sparse_true_divide(self, device, dtype):
         casting_result = dividend.to(torch.get_default_dtype()) / 2
         self.assertEqual(casting_result, torch.true_divide(dividend_sparse, 2).to_dense())
 
+    @float_double_default_dtype
+    def test_reductions(self, device):
+        for fn, meth in ((torch.mean, torch.Tensor.mean),
```

Comment:
Are there problems with ONNX export and/or the TorchScript shape/dtype propagation? I remember that things like torch.mean are hardcoded to return a Tensor of the same dtype. It might be nice to have tests to check if the behavior of those match or not

Reply:
Yep! I'm updating the PR with ONNX support. This was just for initial testing.

Reply:
(Added [WIP] to name, thanks for reminding me!)

Reply:
Recording the results of our autograd-jit conversation here: recommendation is to update the JIT's shape propagation pass to reflect this changes and test by building and running existing tests. There does not seem to be a good way to test shape prop changes and the JIT team would like to remove the pass.